File: cpp/anagram/242.Valid_Anagram.cpp
// leercode Problem 242 : Valid Anagram

#include<iostream>
using namespace std;

// Using unordered_map

bool isAnagram(string s, string t){
	if(s.size() != t.size()){
		return false;
	}
	
	map<char, int> freqS;
	map<char, int> freqT;

	for(int x: s){
		freqS[x]++;
	}
	for(int y: t){
		freqT[y]++;
	}

	return freqS == freqT;

}

int main(){
	isAnagram("anagram", "nagaram");
	return 0;
}


// Runtime: 2 ms | Beats 56.22 %
// Memory: 9.92 | Beats: 56.32 %

File: python/anagram/242.Valid_Anagram.py
# leercode Problem 242 : Valid Anagram


# Using Hashmap
def isAnagram(s, t):
	if len(s) != len(t):
        return False
    freq = {}
    for x, y in zip(s, t):
        freq[x] = freq.get(x, 0) + 1
        freq[y] = freq.get(y, 0) - 1
        
    for count in freq.values():
        if count != 0:
            return False
    return True


# Runtime: 13ms | Beats 47.96 %
# Memory: 17.66 MB | Beats 97.44 %